### PR TITLE
Info collector

### PR DESCRIPTION
**Pull request by s-soroosh**

### CODE REVIEW

File: fandogh_cli/base_commands.py [hunk 1]
```diff
@@ -19,16 +20,19 @@ def invoke(self, ctx):
...
                 exp.response.content))
             click.echo(format_text(exp.message, TextStyle.FAIL), err=True)
             exit(1)
         except VersionException as exp:
+            collect(self, ctx, exp)
             click.echo(format_text("New Version of {} is available, please update to continue "
                                    "using Fandogh services using : `pip install {} --upgrade`".format(NAME, NAME),
                                    TextStyle.FAIL))
         except Exception as exp:
+            collect(self, ctx, exp)
```
Comment (MahdiZareie) [inline fandogh_cli/base_commands.py]: Everyone(other softwares) ask for permission before collect anonymous statistics and stuff, shouldn't we?

File: fandogh_cli/info_collector.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+from fandogh_cli.fandogh_client.error_report_client import report
+
+_static_info = {
+    'python_version': platform.sys.version,
+    'platform': platform.platform(),
+    'cli_version': VERSION
+}
+
+
+def collect(cmd, ctx, exception=None):
```
Comment (MahdiZareie) [inline fandogh_cli/info_collector.py:17]: we can consider doing this inside another thread , it would make the CLI more responsive.

**Review by MahdiZareie — CHANGES_REQUESTED** [review]

File: fandogh_cli/base_commands.py (hunk 1 above)
Comment (s-soroosh) [reply fandogh_cli/base_commands.py]: I was going to do it. but we are just reporting the errors, not tracking the user behavior, I believe it's fine to not ask for permission for such info.

File: fandogh_cli/info_collector.py (hunk 2 above)
Comment (s-soroosh) [reply fandogh_cli/info_collector.py:17]: not really. as CLI process stops on every command. so we cannot guarantee the request has sent to the server before app termination unless we use non daemon thread

**Review by MahdiZareie — APPROVED** [review]

**Pull request merged**